staging/src/k8s.io/client-go/examples/README.md

```diff
@@ -0,0 +1,36 @@
+# client-go Usage Examples
+
+This directory contains examples that cover examples of various use cases
+and functionality for Go client library for Kubernetes.
+
+### Authentication
+
+- [**Authenticate in cluster**](./authenticate-in-cluster)
+- [**Authenticate out of cluster**](./authenticate-out-of-cluster)
+
+### Basics
+
+- [**Managing resources with API**](./create-update-delete-deployment): Create,
+  get, update, delete a Deployment resource.
+
+### Advanced Concepts
+
+- [**Third-party resources**](./third-party-resources): Register a custom
+  resource type with the API, create/update/query this custom type, and write a
+  controller to handle events for this custom resource type.
+- [**Work queues**](./workqueue): Create a custom controller with a Watch for Pod events,
```

Comment:
tl;dr
Consider introduce informer in the description of the tpr example:
"The example demonstrates how to write a controller using the [informer](https://godoc.org/k8s.io/client-go/tools/cache#NewInformer)."

And update the description of workqueue example to:
"Combine the workqueue with the informer to write a controller. Workqueue makes sure the controller doesn't hot-loop on failed operations."

---

> hotloop-free controller

I'm referring to this [line](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/examples/workqueue/main.go#L106), this makes sure the controller to exponential backoff before process the item again. Preventing hot-loop is a main feature of the workqueue.

IMO another important functionality of the workqueue is that it allow multiple workers to work in parallel, unfortunately the example doesn't show that.

> informer framework

I mean the "k8s.io/client-go/tools/cache" package, which is also used in the thirdparty resource example, so we don't need to emphasis "informer framework" in the workqueue example.


> combine the workqueue with a cache to a full controller

Here the "cache" means "k8s.io/client-go/tools/cache".

> synchronize the controller on startup

It refers to this [line](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/examples/workqueue/main.go#L126). It's less important. Probably we should copy this line to the tpr example.